AcademyResidentInformationApi/V1/Infrastructure/Migrations/20200915085446_AddInitialIndexes.cs

```diff
@@ -0,0 +1,42 @@
+using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace AcademyResidentInformationApi.Migrations
+{
+    public partial class AddInitialIndexes : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.Sql("CREATE EXTENSION IF NOT EXISTS pg_trgm;");
+            migrationBuilder.Sql("CREATE INDEX hbmember_forename_idx ON dbo.hbmember USING btree (forename);");
+            migrationBuilder.Sql("CREATE INDEX hbmember_surname_idx ON dbo.hbmember USING btree (surname);");
+            migrationBuilder.Sql("CREATE INDEX hbmember_forename_ilike_idx ON dbo.hbmember USING gin (forename gin_trgm_ops);");
+            migrationBuilder.Sql("CREATE INDEX hbmember_surname_ilike_idx ON dbo.hbmember USING gin (surname gin_trgm_ops);");
+            migrationBuilder.Sql("CREATE INDEX hbhousehold_post_code_idx ON dbo.hbhousehold USING gin (post_code gin_trgm_ops);");
+            migrationBuilder.Sql("CREATE INDEX hbhousehold_addr1_idx ON dbo.hbhousehold USING gin (addr1 gin_trgm_ops);");
+            migrationBuilder.Sql("CREATE INDEX hbhousehold_to_date_idx ON dbo.hbhousehold USING btree (to_date);");
+            migrationBuilder.Sql("CREATE INDEX hbhousehold_postcode_whitespace_replaced_idx ON dbo.hbhousehold USING gin (replace((post_code)::text, ' '::text, ''::text) gin_trgm_ops);");
+            migrationBuilder.Sql("CREATE INDEX hbhousehold_address_whitespace_replaced_idx ON dbo.hbhousehold USING gin (replace((addr1)::text, ' '::text, ''::text) gin_trgm_ops);");
+            migrationBuilder.Sql("CREATE INDEX uc_hbmember ON dbo.hbmember USING btree (claim_id, house_id);");
+            migrationBuilder.Sql("CREATE INDEX ctoccupation_account_ref ON dbo.ctoccupation USING btree (account_ref);");
+            migrationBuilder.Sql("CREATE INDEX ctoccupation_vacation_date ON dbo.ctoccupation USING btree (vacation_date);");
+
+        }
+
+        protected override void Down(MigrationBuilder migrationBuilder)
```

Comment:
Just wondering.

In EF, are there any migration methods that combine the`Up` and `Down` functionality. e.g in rails, you can sometimes use the `Change` method that performs the `Up` function when you run the migration and when you rollback, it performs the `down` function. 

So rather than having a separate `Up` and `Down`, you have one `Change` that knows what to do depending on the circumstance.

Is there something similar in EF?

Reply:
Not that I have ever come across. If I remember correctly, in ruby it will work out the down part from the up so you don't have to write out the down?

Reply:
Yeah, that what I recall. 
So you would only need to describe one action for the migration and if you needed to rollback, it would work out how to perform the opposite but I've seen examples in ruby that still declare both up and down too.